arangod/Aql/IndexExecutor.cpp

```diff
@@ -518,3 +553,56 @@ std::pair<ExecutionState, IndexStats> IndexExecutor::produceRows(OutputAqlItemRo
     }
   }
 }
+
+std::tuple<ExecutionState, IndexExecutor::Stats, size_t> IndexExecutor::skipRows(size_t toSkip) {
+  TRI_IF_FAILURE("IndexExecutor::skipRows") {
+    THROW_ARANGO_EXCEPTION(TRI_ERROR_DEBUG);
+  }
+
+  IndexStats stats{};
+
+  if (_done) {
+    return {ExecutionState::DONE, stats, 0};
+  }
+
+  while (_returned < toSkip) {
+    if (!_input) {
+      if (_state == ExecutionState::DONE) {
+        _done = true;
+        return {_state, stats, _returned};
+      }
+
+      std::tie(_state, _input) = _fetcher.fetchRow();
+
+      if (_state == ExecutionState::WAITING) {
+        return {_state, stats, 0};
+      }
+
+      if (!_input) {
+        TRI_ASSERT(_state == ExecutionState::DONE);
+        _done = true;
+        return {_state, stats, _returned};
+      }
+
+      if (!initIndexes(_input)) {
+        _input = InputAqlItemRow{CreateInvalidInputRowHint{}};
+        continue;
+      }
+    }
+
+    // Skip the next elements from the indexes
+    setIndexesExhausted(!skipIndex(toSkip, stats));
+
+    if (_indexesExhausted) {
+      _input = InputAqlItemRow{CreateInvalidInputRowHint{}};
+      continue;
+    }
+  }
+
+  if (getCursor() != nullptr && getCursor()->hasMore()) {
+    return {ExecutionState::HASMORE, stats, _returned};
+  } else {
+    _done = true;
+    return {ExecutionState::DONE, stats, _returned};
```

Comment:
I think this must be
```suggestion
    return {_state, stats, _returned};
```